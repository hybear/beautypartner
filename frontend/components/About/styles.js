import styled from 'styled-components';

export const Content = styled.main`
  display: grid;
  grid-template:
    'profile artwork' auto
    'about artwork' 10vh
    'design development' 60vh / 60% 40%;
  padding: 40px;
`;

export const Profile = styled.div`
  grid-area: profile;
  display: grid;
  align-items: center;
  grid-template-columns: 25% 75%;
  border-bottom: solid 1px ${({ theme }) => theme.color.lightGrey};
  padding-bottom: 40px;
  .picture {
    margin: 0 auto;
    max-height: 180px;
    max-width: 180px;
    border-radius: 100px;
    object-fit: cover;
  }
  .content {
    margin-left: 20px;
    &__sub {
      font-size: 1em;
      i {
        color: ${({ theme }) => theme.color.primary};
      }
    }
    &__name {
      font-size: 1.875em;
      color: ${({ theme }) => theme.color.primary};
      font-family: 'QuadraSans-Bold', sans-serif;
      margin: 5px 0 10px;
      small {
        font-size: 0.5em;
        font-family: 'QuadraSans-Regular', sans-serif;
      }
    }
    &__shortinfo {
      font-size: 1em;
    }
    &__social {
    }
  }
`;

export const Artwork = styled.div``;
export const About = styled.div``;
export const Design = styled.div``;
export const Development = styled.div``;
