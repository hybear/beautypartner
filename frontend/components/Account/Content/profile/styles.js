import styled from 'styled-components';
import { Text } from '../../../General';
import { Button as GButton } from '../../../General/Form';

export const ProfileContainer = styled.div`
  align-items: start;
  display: grid;
  grid-template:
    'title title' auto
    'profile password' auto
    / 60% 1fr;
  grid-gap: 0 20px;
`;

export const Profile = styled.div`
  background: ${({ theme }) => theme.color.lightGrey};
  border-radius: 0.2rem;
  color: ${({ theme }) => theme.color.grey};
  box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
  display: grid;
  grid-area: profile;
  ${props => !props.handleProfile && `grid-template-columns: repeat(auto-fill, minmax(calc(100% / 3), 1fr))`};
  grid-gap: 5px;
  /* margin: 10px 0; */
  padding: 20px;
  position: relative;
  p {
    font-family: 'QuadraSans-Bold';
  }
`;

export const ProfileItem = styled.span`
  margin: 5px;
  p {
    margin-bottom: 5px;
    font-size: 1.2em;
  }
`;

export const Password = styled.div`
  form {
    align-items: start;
    display: grid;
    grid-template: 'email email' auto;
    grid-gap: 10px;
  }
`;

export const EmailReset = styled(Text)`
  grid-area: email;
  color: ${({ theme, success }) => success && theme.color.success};
  b {
    font-family: 'QuadraSans-Bold';
  }
`;

export const PasswordContainer = styled.div`
  grid-area: password;
  max-width: 30rem;
  padding: 20px;
  /* margin: 10px 0; */
  border-radius: 0.2rem;
  background: ${({ theme }) => theme.color.lightGrey};
  box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
  color: ${({ theme }) => theme.color.grey};
  position: relative;
  > p {
    font-family: 'QuadraSans-Bold';
    font-size: 1.2em;
  }
`;

export const Button = styled(GButton)`
  ${({ profile }) => profile && `margin: 20px 10px 0 0;`}
`;
