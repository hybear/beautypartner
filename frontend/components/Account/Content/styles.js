import styled from 'styled-components';
import { Title as GTitle } from '../../General';

export const Content = styled.div`
  grid-area: cont;
  margin: 20px 0 0 40px;
  padding: 0 20px 0;
  overflow: auto;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: ${({ theme }) => theme.color.white};
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: ${({ theme }) => theme.color.white};
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.color.lightGrey};
  }

  ${({ theme }) => theme.media.tablet`
      margin: 20px 0 80px;
  `}
`;

export const Title = styled(GTitle)`
  text-align: initial;
  font-size: 1.875em;
  font-family: 'QuadraSans-Bold';
  color: ${({ theme }) => theme.color.grey};
  margin-bottom: 20px;
`;
