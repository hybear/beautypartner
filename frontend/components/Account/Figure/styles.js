import styled, { keyframes } from 'styled-components';

export const FigureContainer = styled.div`
  grid-area: figures;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  ${({ theme }) => theme.media.tablet`
    display: none;
  `}
`;

const bright = keyframes`
  0% {
    filter: contrast(80%);
  }

  50% {
    filter: contrast(120%);
  }

  100% {
    filter: contrast(80%);
  }
`;

export const Image = styled.img`
  mask-image: url('/assets/icon/MaskAccount.svg');
  mask-position: right;
  mask-repeat: no-repeat;
  mask-size: cover;
  object-fit: cover;
  object-position: right;
  height: 100%;
  width: 100%;
  animation: ${bright} 3s linear infinite;
`;
