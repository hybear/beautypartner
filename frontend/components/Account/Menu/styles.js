import styled from 'styled-components';
import { Avatar as IAvatar } from '../../../public/assets/icon/Avatar.svg';

export const Drawer = styled.div`
  grid-area: nav;
  width: 100%;
  display: grid;
  grid-template:
    'avatar' 15%
    'balance' 8%
    'list' auto
    'signout' 5%
    / 100%;
  grid-row-gap: 1em;
  padding-top: 40px;
  background: #fafafa;
  border-right: solid 1px ${({ theme }) => theme.color.lightGrey};
`;

export const AvatarContainer = styled.div`
  grid-area: avatar;
  align-items: center;
  display: flex;
  height: 100%;
  width: 100%;
  .avatar {
    align-items: center;
    display: flex;
    margin-left: 1.5em;
    color: ${({ theme }) => theme.color.grey};
    .avatar__name {
      display: flex;
      flex-flow: column;
      font-family: 'QuadraSans-Bold', sans-serif;
      font-size: 1.875rem;
      margin-left: 10px;
      small {
        font-family: 'QuadraSans-Regular', sans-serif;
        font-size: 0.75em;
      }
    }
  }
`;

export const List = styled.ul`
  grid-area: list;
`;

export const ListItem = styled.li`
  background: ${({ theme, active }) => (active ? theme.color.primary : 'transparent')};
`;

export const Avatar = styled(IAvatar)`
  background: ${({ theme }) => theme.color.white};
  border-radius: 60px;
  display: flex;
  min-height: 6rem;
  min-width: 6rem;
  overflow: hidden;
  box-shadow: inset 2px 3px 3px 0px rgba(0, 0, 0, 0.25);
  padding: 10px;
`;

export const Balance = styled.span`
  grid-area: 'balance';
  align-items: flex-end;
  background: ${({ theme }) => theme.color.primary};
  border-radius: 0.2em;
  color: ${({ theme }) => theme.color.white};
  display: flex;
  flex-flow: column;
  font-size: 1.25rem;
  font-family: QuadraSans-Bold;
  justify-content: center;
  left: -50%;
  position: relative;
  padding: 0 1.875rem;
  small {
    font-family: QuadraSans-Regular;
    font-size: 0.75em;
  }
`;

export const SignOutContainer = styled.div`
  grid-area: signout;
  width: 90%;
  display: flex;
  margin: 0 auto;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
`;
