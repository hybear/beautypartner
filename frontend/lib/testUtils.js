import casual from 'casual';

// seed it so we get consistent results
casual.seed(777);

const fakeItem = () => ({
  __typename: 'Item',
  id: 'abc123',
  listPrice: 180000,
  bestPrice: 155200,
  category: 100,
  reference: '15523',
  title: 'LA MER The Moisturizing Soft Cream',
  description:
    'An ultra-soft face moisturizer that nurtures for beautiful, glowing radiance, leaving skin looking naturally brightened, hydrated, and restored.',
  image: 'laMer.jpg',
  largeImage: 'largelaMer.jpg',
});

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  birthday: '1996-03-08T14:18:00.000Z',
  creditCards: [],
  phone: casual.phone,
  balance: 1800,
  orders: [],
  cart: [],
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: `${casual.word}.jpg`,
  title: casual.words(),
  listPrice: 4234,
  bestPrice: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 400,
  cashback: 5,
  paymentMethod: 'stripe',
  items: [fakeOrderItem(), fakeOrderItem()],
  status: ['APROVED'],
  createdAt: '2020-02-17T01:52:31.269Z',
  user: fakeUser(),
});

const fakeCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  item: fakeItem(),
  user: fakeUser(),
  ...overrides,
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export { LocalStorageMock, fakeItem, fakeUser, fakeCartItem, fakeOrder, fakeOrderItem };
