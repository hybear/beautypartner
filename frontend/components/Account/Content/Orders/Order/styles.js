import styled from 'styled-components';
import { Title } from '../../styles';
import { Button } from '../../../../General/Form';

export const OrderStylesContainer = styled.div`
  height: 100%;
  overflow: hidden;
`;

export const OrderTotal = styled.p`
  font-size: 1.1em;
  margin: 0 0 15px;
`;
export const OrderListTitle = styled.div`
  /* display: grid;
  grid-template-columns: repeat(5, auto);
  text-align: center; */
`;
export const OrderStyles = styled.li`
  margin: 10px 0;
  display: block;
  .order {
    display: grid;
    justify-content: space-between;
    align-items: center;
    grid-template-columns: repeat(4, auto);
    padding: 20px;
    transition: all 0.2s;
    text-align: center;
    color: ${({ theme }) => theme.color.grey};
    text-decoration: none;
    background: ${({ theme }) => theme.color.lightGrey};
    box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
    border-radius: 0.2rem;

    .order__title {
      font-family: 'QuadraSans-Bold', sans-serif;
      margin-bottom: 10px;
      font-size: 1.25em;
    }

    .order__info {
      &.cashback {
        small {
          display: block;
          background: ${({ theme }) => theme.color.primary};
          color: ${({ theme }) => theme.color.white};
          padding: 0.1em 0;
          border-radius: 2px;
        }
      }
    }
  }
`;

export const OrderButton = styled(Button)`
  a {
    color: ${({ theme }) => theme.color.white};
    &:hover {
      text-decoration: none;
    }
  }
`;

export const OrderList = styled.ul`
  overflow-x: hidden;
  overflow-y: auto;
  margin-bottom: 20px;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.grey};
  }
`;
