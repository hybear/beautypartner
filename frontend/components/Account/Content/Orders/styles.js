import styled from 'styled-components';
import { Title } from '../styles';

export const OrderStylesContainer = styled.div`
  height: 100%;
  overflow: hidden;
`;

export const OrderTotal = styled.p`
  font-size: 1.1em;
  margin: 0 15px 20px;
`;

export const OrderList = styled.ul`
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  padding-bottom: 40px;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.grey};
  }
`;
