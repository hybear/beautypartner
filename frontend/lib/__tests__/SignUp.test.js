import { MockedProvider } from '@apollo/react-testing';
import wait from 'waait';
import SignUp, { SIGNUP_MUTATION } from '../../components/SignUp';
import { CURRENT_USER_QUERY } from '../../components/User';
import { fakeUser } from '../testUtils';
import { act } from 'react-dom/test-utils';
import { ApolloConsumer } from 'react-apollo';
import Router from 'next/router';
import { ThemeProviderWrapper } from '../../styles/theme';

Router.router = {
  push() {},
};

const me = fakeUser();
const mocks = [
  {
    request: {
      query: SIGNUP_MUTATION,
      variables: {
        name: me.name,
        email: me.email,
        password: 'test22',
        document: '123456789',
      },
    },
    result: {
      data: {
        signup: {
          id: 'abc123',
          name: me.name,
          email: me.email,
          document: '123456789',
          avatar: '1',
          __typename: 'User',
        },
      },
    },
  },
  {
    request: {
      query: CURRENT_USER_QUERY,
    },
    result: { data: { me } },
  },
];

function type(wrapper, name, value) {
  wrapper.find(`input[name="${name}"]`).simulate('change', {
    target: { name, value },
  });
  // console.log(wrapper.find(`input[name="${name}"]`).debug())
}

describe('<SignUp />', () => {
  it('renders and matches snapshots', async () => {
    const wrapper = mount(
      <MockedProvider>
        <SignUp />
      </MockedProvider>,
      {
        wrappingComponent: ThemeProviderWrapper,
      }
    );
    await wait();
    wrapper.update();
    expect(wrapper.find('form')).toMatchSnapshot();
  });

  it('calls the mutation properly', async () => {
    await act(async () => {
      let apolloClient;
      const wrapper = mount(
        <MockedProvider mocks={mocks} addTypename={false}>
          <ApolloConsumer>
            {client => {
              apolloClient = client;
              return <SignUp />;
            }}
          </ApolloConsumer>
        </MockedProvider>,
        {
          wrappingComponent: ThemeProviderWrapper,
        }
      );
      await wait();
      wrapper.update();

      type(wrapper, 'fullname', me.name);
      type(wrapper, 'email', me.email);
      type(wrapper, 'password', 'test22');
      type(wrapper, 'document', '123456789');

      wrapper.update();
      wrapper.find('form').simulate('submit');
      await wait();
      const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
      console.log(user);
    });
  });
});
