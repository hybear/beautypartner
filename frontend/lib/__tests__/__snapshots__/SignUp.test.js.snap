// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders and matches snapshots 1`] = `
<form
  className="Form-omwzro-0 styles__Form-pihe1-1 idMdxA"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="form__container"
  >
    <div
      className="form__grid"
    >
      <Label
        htmlFor="fullname"
      >
        <label
          className="Label-sc-2yv4fq-0 bFjpBR"
          htmlFor="fullname"
        >
          Full Name*
        </label>
      </Label>
      <Input
        checked={false}
        id="fullname"
        name="fullname"
        onChange={[Function]}
        placeholder="Ex: Lydia Hallie"
        required={true}
        tabindex="1"
        type="text"
        value=""
      >
        <input
          checked={false}
          className="Input-sc-145vwqf-0 kWeKyo"
          id="fullname"
          name="fullname"
          onChange={[Function]}
          placeholder="Ex: Lydia Hallie"
          required={true}
          type="text"
          value=""
        />
      </Input>
    </div>
    <div
      className="form__grid"
    >
      <Label
        htmlFor="email"
      >
        <label
          className="Label-sc-2yv4fq-0 bFjpBR"
          htmlFor="email"
        >
          E-mail*
        </label>
      </Label>
      <Input
        checked={false}
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Ex: lydia.hallie@email.com"
        required={true}
        tabindex="2"
        type="email"
        value=""
      >
        <input
          checked={false}
          className="Input-sc-145vwqf-0 kWeKyo"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Ex: lydia.hallie@email.com"
          required={true}
          type="email"
          value=""
        />
      </Input>
    </div>
    <div
      className="form__grid"
    >
      <Label
        htmlFor="document"
      >
        <label
          className="Label-sc-2yv4fq-0 bFjpBR"
          htmlFor="document"
        >
          CPF*
        </label>
      </Label>
      <InputElement
        mask="999.999.999-99"
        onChange={[Function]}
        value=""
      >
        <Input
          checked={false}
          id="document"
          name="document"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          placeholder="000.000.000-00"
          required={true}
          tabindex="3"
          type="text"
          value=""
        >
          <input
            checked={false}
            className="Input-sc-145vwqf-0 kWeKyo"
            id="document"
            name="document"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            placeholder="000.000.000-00"
            required={true}
            type="text"
            value=""
          />
        </Input>
      </InputElement>
    </div>
    <div
      className="form__grid"
    >
      <Label
        htmlFor="password"
      >
        <label
          className="Label-sc-2yv4fq-0 bFjpBR"
          htmlFor="password"
        >
          Password*
        </label>
      </Label>
      <Input
        checked={false}
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="*******"
        required={true}
        tabindex="4"
        type="password"
        value=""
      >
        <input
          checked={false}
          className="Input-sc-145vwqf-0 kWeKyo"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="*******"
          required={true}
          type="password"
          value=""
        />
      </Input>
      <Info
        hidden={false}
      >
        <span
          className="Info-sc-19b1ctt-0 etcRNQ"
          hidden={false}
        >
          <ul>
            <li>
              Uppercase
            </li>
            <li>
              Lowercase
            </li>
            <li>
              Number
            </li>
            <li>
              Special Char
            </li>
            <li>
              Min 8 characters
            </li>
          </ul>
        </span>
      </Info>
    </div>
  </div>
  <styles__Button
    disabled={true}
    fullWidth={true}
    primary={true}
    type="submit"
  >
    <button
      className="Button-sc-1hbypwi-0 styles__Button-pihe1-4 bHNFfx"
      disabled={true}
      type="submit"
    >
      Sign
       Up
    </button>
  </styles__Button>
  <styles__Text>
    <p
      className="Text-sc-1mkzdb2-0 styles__Text-pihe1-2 fhTDbb"
    >
      Already have an account?
      <Link
        href="/signin"
      >
        <a
          href="/signin"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           Sign in
        </a>
      </Link>
    </p>
  </styles__Text>
</form>
`;
